pkgs/top-level/stage.nix

```diff
@@ -9,8 +9,40 @@
    import `pkgs/default.nix` or `default.nix`. */
 
 
-{ # The system (e.g., `i686-linux') for which to build the packages.
-  system
+{ # Misc parameters kept the same for all stages
+  #
+
+  # Utility functions, could just import but passing in for efficiency
+  lib
+
+, # Use to reevaluate Nixpkgs; a dirty hack that should be removed
```

Comment:
This *isn't* used for normal bootstrapping. calling this doesn't just eval one stage, but does a whole fresh rebootstrap of the entire thing. currently used for i686-forced packages, and getting build tools for some arm ram flasher thingy.

The former is kind a x86-specific hack. With this we could do a real cross-compile but then we wouldn't share hashes with native x86_32. The latter is interesting, and implies maybe there should be some "lightweight" way to get cross tools without creating a bootstrapping stage? I dunno I say future work on both counts.